### PR TITLE
More documentation movement and cleanup for linalg

### PR DESCRIPTION
**Pull request by kshyatt**
Removed square brackets from signatures, added
more links, clarified some wording.

### CODE REVIEW

**Comment by kshyatt** [comment]
@andreasnoack or @tkelman can I get an OK to merge?

File: base/linalg/ldlt.jl [hunk 1]
```diff
@@ -17,9 +17,9 @@ convert{T,S,U}(::Type{Factorization{T}}, F::LDLt{S,U}) = convert(LDLt{T,U}, F)
 # SymTridiagonal
 """
```
Comment (tkelman) [inline base/linalg/ldlt.jl]: unnecessary blank line here?

File: base/sparse/spqr.jl [hunk 2]
```diff
@@ -136,6 +136,15 @@ function qmult{Tv<:VTypes}(method::Integer, QR::Factorization{Tv}, X::Dense{Tv})
...
 
+
+"""
+    qrfact(A) -> SPQR.Factorization
+
+Compute the `QR` factorization of a sparse matrix `A`. A fill-reducing permutation is used.
+The main application of this type is to solve least squares problems with `\\`. The function
+calls the C library SPQR and a few additional functions from the library are wrapped but not
+exported.
+"""
```
Comment (tkelman) [inline base/sparse/spqr.jl]: attach to the other signature?

File: base/sparse/cholmod.jl [hunk 3]
```diff
@@ -1313,7 +1316,7 @@ To include the effects of permutation,
 it's typically preferable to extact "combined" factors like `PtL = F[:PtL]`
```
Comment (tkelman) [inline base/sparse/cholmod.jl]: existing typo, s/extact/extract/

File: base/sparse/cholmod.jl [hunk 4]
```diff
@@ -1399,8 +1404,9 @@ Compute the ``LDL'`` factorization of a sparse matrix `A`.
...
 A fill-reducing permutation is used. `F = ldltfact(A)` is most frequently
 used to solve systems of equations `A*x = b` with `F\\b`. The returned
-factorization object `F` also supports the methods `diag`,
-`det`, and `logdet`. You can extract individual factors from `F` using `F[:L]`.
+factorization object `F` also supports the methods [`diag`](:func:`diag`),
+[`det`](:func:`det`), and [`logdet`](:func:`logdet`).
+You can extract individual factors from `F` using `F[:L]`.
 However, since pivoting is on by default, the factorization is internally
 represented as `A == P'*L*D*L'*P` with a permutation matrix `P`;
 using just `L` without accounting for `P` will give incorrect answers.
```
Comment (tkelman) [inline base/sparse/cholmod.jl]: on the next line, there's another existing typo, s/extact/extract/

**Review by tkelman — CHANGES_REQUESTED** [review]
github has review tools now, testing https://github.com/blog/2256-a-whole-new-github-universe-announcing-new-tools-forums-and-features

**Comment by kshyatt** [comment]
@andreasnoack I've dealt with the `bkfact` thing here

**Review by tkelman — APPROVED** [review]

File: base/linalg/bunchkaufman.jl [hunk 5]
```diff
@@ -57,7 +57,7 @@ end
 """
     bkfact(A, uplo::Symbol=:U, symmetric::Bool=issymmetric(A), rook::Bool=false) -> BunchKaufman
 
-Compute the Bunch-Kaufman [^Bunch1977] factorization of a real symmetric or complex Hermitian
+Compute the Bunch-Kaufman [^Bunch1977] factorization of a symmetric or complex Hermitian
```
Comment (tkelman) [inline base/linalg/bunchkaufman.jl]: maybe add a comma? `symmetric, or complex Hermitian,`

Comment (andreasnoack) [inline base/linalg/bunchkaufman.jl]: Great. I agree with @tkelman that the `complex` part can go.

Comment (kshyatt) [inline base/linalg/bunchkaufman.jl]: Ok, I'll take that off and `[ci skip]`?

Comment (tkelman) [inline base/linalg/bunchkaufman.jl]: be sure to do it both places, but sure

**Pull request merged**